src/changelog/comment_changes.rs

```diff
@@ -280,7 +287,29 @@ impl Logic {
             },
             |r| {
                 self.repository = Some(r);
-                Ok(())
+
+                if let Some(tag) = &self.cli.tag {
+                    if let Some(repository) = &self.repository {
+                        if let Ok(target) =
+                            repository.resolve_reference_from_short_name(tag)
+                        {
+                            if target.is_tag() {
+                                self.stop_at_tag_oid = target.target();
+                                Ok(())
+                            } else {
+                                eprintln!("{tag} does not seem to be a tag.");
+                                Err(ExitCode::Usage)
+                            }
+                        } else {
+                            eprintln!("Tag {tag} does not seem to exist.");
+                            Err(ExitCode::Usage)
+                        }
```

Comment:
It might happen that the tag name contains typos, so this edge case needs to be caught.